apps/worker/src/app/workflow/usecases/send-message/send-message.usecase.ts

```diff
@@ -74,19 +77,27 @@ export class SendMessage {
     private subscriberRepository: SubscriberRepository,
     private tenantRepository: TenantRepository,
     private analyticsService: AnalyticsService,
+    private normalizeVariablesUsecase: NormalizeVariables,
     protected moduleRef: ModuleRef
   ) {
     this.resonateUsecase = requireInject('resonate', this.moduleRef);
   }
 
   @InstrumentUsecase()
-  public async execute(command: SendMessageCommand) {
+  public async execute(command: SendMessageCommand): Promise<{ status: 'success' | 'canceled' }> {
     const payload = await this.buildCompileContext(command);
 
-    const [shouldRun, preferred] = await Promise.all([
-      this.filter(command, payload),
-      this.filterPreferredChannels(command.job),
-    ]);
+    const variables = await this.normalizeVariablesUsecase.execute(
```

Comment:
the reason for the separation of variable normalize and filter.
we need to pass the normalized variables to `resonate` use case, if the step is skipped we have no reason to execute the `filter` function. by that we will reduce the possible compute and IO operation.